### PR TITLE
adding files for FSQ DQM code

### PR DESCRIPTION
**Pull request by anmehta**
First Version of FSQ DQM code

### CODE REVIEW

**Comment by anmehta** [comment]
scram build code-checks

**Comment by prebello** [comment]
Hi @anmehta let's try the mentioned changes and push them.

**Comment by jfernan2** [comment]
please test

**Comment by prebello** [comment]
Dear @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien, 
@anmehta  will test few wfs to get root files with enough statistics to approve distributions. 
Please your review should come later. Thank you

**Comment by prebello** [comment]
Hi @anmehta could you please check the errors? 
you go to the summary link where you can search logs for each relval to check what is going on. Maybe DQM experts can help you to understand what is missing.

**Comment by prebello** [comment]
please test workflow 10823,136.837

**Comment by prebello** [comment]
please test workflow 10823,136.837

**Comment by prebello** [comment]
Dear DQM and ORP teams,  @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien,  @fabiocos 

test was approved and the FSQ DQM code is ready to be reviewed and hopefully merged in the next pre release if possible, acc the FSQ PAG validator . 

May we kindly ask your attention to it?

Thank you in advance.

**Comment by davidlange6** [comment]
hi- I was looking at this PR for other reasons, and had 3 suggestions:

1) avoid the duplicated getByLabel call. One time is enough.
2) use the standard cms deltaPhi (you include it already)
3) avoid the copy/type conversion and sort of track 4-momenta just for the reason of finding the phi of the highest pt track.

**Comment by anmehta** [comment]
Hi @davidlange6,
thanks for your suggestions.
Could you please elaborate your first comment. I could not get this. 
I fixed the second comment. For the third we use the leading track as a reference so I  need both its pT as well. That's why we sort the collection. Is there any way out to avoid this. 
Please suggest.

Thanks

**Comment by davidlange6** [comment]
> On May 16, 2018, at 1:27 PM, anmehta <notifications@github.com> wrote:
> 
> Hi @davidlange6,
> thanks for your suggestions.
> Could you please elaborate your first comment. I could not get this.

combine these into one getbylabel call (there are several examples)

if(!iEvent.getByToken(pvs_, privtxs)) return;
iEvent.getByToken(pvs_, privtxs);

> I fixed the second comment. For the third we use the leading track as a reference so I need both its pT as well. That's why we sort the collection. Is there any way out to avoid this.

just loop over tracks to identify the index of the one with the highest pt.

**Comment by dmitrijus** [comment]
+1

**Comment by fabiocos** [comment]
please test

**Comment by dmitrijus** [comment]
+1

File: DQM/Physics/src/FSQDQM.cc [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+         recoCastorJets.clear();
+	 edm::Handle<BasicJetCollection> castorJets;
+	 //	 if(!iEvent.getByToken(tok_castorjet_, castorJets)) return;
+	 iEvent.getByToken(tok_castorjet_, castorJets);
+	 if(castorJets.isValid()){
+	 for (unsigned ijet=0; ijet<castorJets->size();ijet++) {
+	   recoCastorJets.push_back((*castorJets)[ijet]);
+	 }
+	 for (unsigned ijet=0; ijet<recoCastorJets.size(); ijet++) {
+	   cout<<recoCastorJets[ijet].pt()<<endl;
```
Comment (fabiocos) [inline DQM/Physics/src/FSQDQM.cc]: @anmehta @dmitrijus this is adding lots of printouts to the output, please eliminate it!

**Comment by jfernan2** [comment]
please test

**Comment by fabiocos** [comment]
+1

the unwanted printout has been masked

**Comment by prebello** [comment]
@fabiocos  could you please let me know if the content of this PR will appear in the next (pre) release? thank you

**Comment by fabiocos** [comment]
@prebello as it has been merged, it will obviously be part of CMSSW_10_2_0_pre5

**Pull request merged**